components/brave_wallet_ui/components/extension/sign-panel/index.tsx

```diff
@@ -166,8 +175,38 @@ function SignPanel (props: Props) {
               text={getLocale('braveWalletSignTransactionMessageTitle')}
             />
           </TabRow>
+
+          {hasUnicode(selectedQueueData.message) &&
+            <WarningBox warningType='warning'>
+              <WarningTitleRow>
+                <WarningIcon color={'warningIcon'} />
+                <WarningTitle warningType='warning'>
+                  {
+                    'Non-ASCII characters detected!' // TODO
+                  }
+                </WarningTitle>
+              </WarningTitleRow>
+              <LearnMoreButton
+                onClick={() => setRenderUnicode(prev => !prev)}
+              >
+                {
+                 renderUnicode
+                  ? 'View decoded message' // TODO: Locale
+                  : 'View encoded message' // TODO: Locale
```

Comment:
to me this would be more clear as `View message in ASCII encoding` and `View original message`

Reply:
updated here: 46a7213